packages/serverless-orchestration/src/ServerlessHub.js

```diff
@@ -145,60 +145,67 @@ hub.post("/", async (req, res) => {
+            resolve(singleChainId);
+          })
+        );
+      }
 
       // If STORE_MULTI_CHAIN_BLOCK_NUMBERS is set then this bot requires to know a number of last seen blocks across
-      // a set of chainIds. Construct a batch promise to evaluate the latest block number for each chainId.
+      // a set of chainIds. Append a batch promise to evaluate the latest block number for each chainId.
       if (configObject[botName]?.environmentVariables?.STORE_MULTI_CHAIN_BLOCK_NUMBERS) {
         const multiChainIds = configObject[botName]?.environmentVariables?.STORE_MULTI_CHAIN_BLOCK_NUMBERS;
-        let promises = [];
         for (const chainId of multiChainIds) {
-          promises.push(
+          // If we've seen this chain ID or it is covered by the singleChainId we can skip it.
+          if (blockNumbersForChain[chainId] || chainId === singleChainId) continue;
+
+          blockNumberPromises.push(
             _getLastQueriedBlockNumber(req.body.configFile, chainId, logger),
-            _getBlockNumberOnChainIdMultiChain(configObject[botName], chainId)
+            _getBlockNumberOnChainIdMultiChain(configObject[botName], chainId),
+            new Promise((resolve) => {
+              resolve(chainId);
+            })
           );
         }
-        let results = await Promise.all(promises);
-        results.forEach((_, index) => {
-          if (index % 2 !== 0) return;
-          const chainId = multiChainIds[Math.floor(index / 2)];
-          blockNumbersForChain[chainId] = {
-            lastQueriedBlockNumber: results[index],
-            latestBlockNumber: results[index + 1],
-          };
-        });
       }
 
-      // If we've seen this chain ID already we can skip it.
-      if (blockNumbersForChain[chainId]) continue;
-
-      // Fetch last seen block for this chain and get the head block for the chosen chain, which we'll use to override the last queried block number
-      // stored in GCP at the end of this hub execution.
-      let [lastQueriedBlockNumber, latestBlockNumber] = await Promise.all([
-        _getLastQueriedBlockNumber(req.body.configFile, chainId, logger),
-        _getLatestBlockNumber(botWeb3),
-      ]);
-
-      // If the last queried block number stored on GCP Data Store is undefined, then its possible that this is
-      // the first time that the hub is being run for this chain. Therefore, try setting it to the head block number
-      // for the chosen node.
-      if (!lastQueriedBlockNumber && latestBlockNumber) {
-        lastQueriedBlockNumber = latestBlockNumber;
-      }
-      // If the last queried number is still undefined at this point, then exit with an error.
-      else if (!lastQueriedBlockNumber)
-        throw new Error(
-          `No block number for chain ID stored on GCP and cannot read head block from node! chainID:${chainId} spokeCustomNodeUrl:${spokeCustomNodeUrl}`
-        );
+      const blockNumberResults = await Promise.all(blockNumberPromises);
```

Comment:
is it possible to continue if some of these calls fail, or should teh whole thing fail if any fail?  also does this call strain providers with many chains or should we be OK?